backend/src/services/wallet_service.py

```
@@ -59,43 +70,56 @@ async def get_wallet_info(self) -> Optional[Dict]:
                     tokens.append({
                         "account_address": str(pubkey),
                         "mint_address": mint,
-                        "balance_raw": amount,
-                        "balance": 0, # Placeholder for balance with decimals
-                        "usd_value": 0.0 # Placeholder
+                        "balance_raw": amount_raw,
+                        "balance": amount,
+                        "decimals": decimals,
+                        "usd_value": 0.0
                     })
                 except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)